hw/ip/aes/pre_sca/alma/cpp/testbench.h

```diff
@@ -0,0 +1,73 @@
+// Copyright IAIK.
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "verilated.h"
+#include "verilated_vcd_c.h"
+
+template <class Module>
+struct Testbench {
+  unsigned long m_tickcount;
+  Module *m_core;
+  VerilatedVcdC *m_trace;
+
+  Testbench(void) {
+    Verilated::traceEverOn(true);
+    m_core = new Module;
+    m_tickcount = 0ul;
+  }
+
+  virtual ~Testbench(void) {
+    delete m_core;
+    m_core = NULL;
+  }
+
+  virtual void opentrace(const char *vcdname) {
+    if (!m_trace) {
+      m_trace = new VerilatedVcdC;
+      m_core->trace(m_trace, 99);
+      m_trace->open(vcdname);
+    }
+  }
+
+  virtual void closetrace(void) {
+    if (m_trace) {
+      m_trace->close();
+      m_trace = NULL;
+    }
+  }
+
+  virtual void reset(void) {
+    m_core->rst_ni = 0;
+    this->tick();
+    this->tick();
+    m_core->rst_ni = 1;
+  }
+
+  virtual void tick(void) {
+    // Falling edge
+    m_core->clk_i = 0;
+    m_core->eval();
+    if (m_trace)
+      m_trace->dump(20 * m_tickcount);
+
+    // Rising edge
+    m_core->clk_i = 1;
+    m_core->eval();
+    if (m_trace)
+      m_trace->dump(20 * m_tickcount + 10);
+
+    // Falling edge settle eval
+    m_core->clk_i = 0;
+    m_core->eval();
+
+    if (m_trace)
+      m_trace->flush();
+    m_tickcount++;
+  }
+
+  virtual bool done(void) { return (Verilated::gotFinish()); }
```

Comment:
Nit: The outer parens aren't necessary (and nor is the `void`):
```suggestion
  virtual bool done() { return Verilated::gotFinish(); }
```

Reply:
Done.